### ISSUE TITLE
Stream continuously fills RAM without getting flushed/cleared

### ISSUE THREAD

**Issue by steebchen** [nodejs/node-v0.x-archive#14189]
When I stream a 500MB file, the RAM continuously decreases, the swap RAM will be used and after another 500MB file the server crashes because it is out of RAM. Do I have to (or can I) manually delete the data object or flush/clear the RAM somehow?

``` javascript
http.get("..", function (response) {
    stream.on("data", function (data) {
        res.write(data);
            // Do I have to delete 'data' somehow?
    });

    stream.on("end", function () {
        res.end();
    });
    response.resume();
    response.pipe(stream);
});
```

**Comment by misterdjules** [comment]
Thanks for filing this issue.

Should it be considered a duplicate of #9360? Also, the code you pasted in the issue's description doesn't run as is, would you mind posting some code that runs without using external dependencies, and without requiring any change?

Thank you!

**Comment by misterdjules** [comment]
We would also need to know which version of node you're using and on what operating system. Thanks again!

**Comment by steebchen** [comment]
I tried a lot of things now - you don't want to know how much hours I wasted. 
What I wanted to add is that all methods I tried (1. the "normal" way, 2. with the request module (link below) WORK ON WINDOWS, BUT NOT ON MY LINUX SERVER.

On Linux it's working with the request module (but not with the code above!) on node v0.8.21 (not with v0.10.35 and not with v0.12.1, tried both!) and the `--expose-gc` arg and the line `global.gc()`. 

I'm using the express module, should I try to rebuild this issue without it?

https://github.com/request/request
